lib/chef_apply/temp_cookbook.rb

```diff
@@ -54,7 +52,8 @@ def from_existing_recipe(existing_recipe_path)
         ChefApply::Log.debug("Found single recipe at path '#{existing_recipe_path}'")
         recipe = File.basename(existing_recipe_path)
         recipe_name = File.basename(existing_recipe_path, ext_name)
-        name = "cw_recipe"
+        @type = :resource
```

Comment:
I think its fine for this PR, but a 'resource type of temporary cookbook' seems like a smell. We could pull out 2 classes?

*edit*

It looks like this type is only used to determine messaging. What if we just stored the message to communicate as an attribute instead? But that seems like it violates the principal of keeping the UI separate...

Reply:
The difference between the two messages is `Converging local recipe %1 on target...` for type `:recipe` and `Converging %1 on target...` for type `:resource`.

Looks like this messaging could be cleaned up in general. What about something like `Converging recipe cookbook_name::recipe_name on target...` for times when we found a cookbook and recipe, 
`Converging recipe /path/to/local/recipe.rb on target...` for when we just fine a single recipe not in a cookbook and `Converging resource resource_type[resource_name] on target...` for when we have a single resource.

Then instead of `:type` as the property it could be `:from` and could have values of `:full_cookbook`, `:single_recipe`, and `:resource`. Thoughts?

Reply:
That would work.  Though re-reading the messages, I'm kind of wondering if we could just drop the descriptor entirely, to : `Converging %1 on target` or `Converging target with %1` , where %1 is formatted as you said.

Or super short-n-sweet, what do you think of  `Applying %1`? Since it's already in the context of a status line with the target host name on it

Reply:
Short-n-sweet FTW! Yeah, if we format the `cookbook.from` appropriately then we can definitely just go with `Applying %1` for all of them. 👍